realm/realm-library/src/main/java/io/realm/internal/ManagedObject.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.internal;
+
+/**
+ * This internal interface represents a java object that corresponds to data managed in the Realm core.
+ * It specifies the operations common to all such objects.
+ */
+public interface ManagedObject {
+
+    /**
+     * Checks to see if this object is managed by Realm..
+     *
+     * @return {@code true} if this is a managed Realm object, {@code false} otherwise.
+     */
+    boolean isManaged();
+
+    /**
+     * Checks tos see if the RealmObject is still valid to use i.e., the RealmObject hasn't been deleted nor has the
```

Comment:
`tos` -> `to`
`the RealmObject` -> `the object` maybe? it is implemented by the `RealmCollection` as well.